test/api/workflows_format_2/converter.py

```diff
@@ -64,16 +92,88 @@ def python_to_workflow(as_python):
         steps = steps_as_dict
 
     for i, step in steps.iteritems():
+        step_type = step.get("type", None)
+        if "run" in step:
+            if step_type is not None:
+                raise Exception("Steps specified as run actions cannot specify a type.")
+            run_action = step.get("run")
+            if "@import" in run_action:
+                if len(run_action) > 1:
+                    raise Exception("@import must be only key if present.")
+
+                run_action_path = run_action["@import"]
+                runnable_path = os.path.join(conversion_context.workflow_directory, run_action_path)
+                with open(runnable_path, "r") as f:
+                    runnable_description = yaml.load(f)
+                    run_action = runnable_description
+
+            run_class = run_action["class"]
+            run_to_step_function = eval(RUN_ACTIONS_TO_STEPS[run_class])
+
+            run_to_step_function(conversion_context, step, run_action)
+            del step["run"]
+
+    for i, step in steps.iteritems():
```

Comment:
Fixes in my CWL/subworkflow branch (https://github.com/common-workflow-language/galaxy/commit/d5933b507eab4ccad2955c1a76ad24a23e984596). It will be in my next round of small backend refactoring.